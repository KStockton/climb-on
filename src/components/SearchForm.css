.SearchForm-header {
 background-image: url('../images/climbon.jpg');
 background-position: 0 -70px;
 background-size: cover;
 background-repeat: no-repeat;
 text-align: center;
 min-width: 100vh;
}

form { 
 min-height: 550px; 
 display: flex;
 flex-direction: row;
 justify-content: center;
 align-items: center;
 font-size: calc(10px + 2vmin); 
 color: rgba(249, 245, 245, 0.851);
} 

.SearchForm-search-bar {
 height: 2.5rem;
 width: 12.3rem;
 background-color: rgba(198, 157, 157, 0.968); 
 border-top-left-radius: 5px;
 border-bottom-left-radius: 5px;
 border-style: none;
 border-color: #c3e5fe;
 font-size: 20px;
 color: rgb(251, 251, 252);
 padding: 4px 0px 0px 11px;
}

button {
 background-color: #54494B;
 color: #ffffff;
 margin-left: -4px;
 height: 2.5rem;
 border-top-right-radius: 8px;
 border-bottom-right-radius: 8px;
 padding: 5px 7px 6px 9px;
 font-size: 27px;
 cursor: pointer; 
 border: none;
}