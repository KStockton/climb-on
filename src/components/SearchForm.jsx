import React, { Component } from 'react';
import './SearchForm.scss';

class SearchForm extends Component {

  constructor() {
    super();
    this.state = {
      inputValue: ''
    };
  }

  _handleChange = e => {
    this.setState({ inputValue: e.target.value });
  } 

  _handleSubmit = e => {
    e.preventDefault();
    this.props.updateSearch(this.state.inputValue);
  }

  render() {
    return (
      <section className="SearchForm">
        <h1 className="logo">Climb<span>On</span></h1>
        <form className="SearchForm-form" onSubmit={this._handleSubmit}>
          <input className="SearchForm-search-bar" type="text" name="search" autoComplete="on" placeholder="Search" onChange={this._handleChange}/>
          <button className="SearchForm-submit-button"><i className="fas fa-search"></i></button>
        </form>
      </section>
    )
  }
}

export default SearchForm;