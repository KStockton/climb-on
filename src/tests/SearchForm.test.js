import React from 'react';
import SearchForm from '../components/SearchForm';
import { shallow } from 'enzyme';

const mock_handleSubmit = jest.fn();

describe('SearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SearchForm submitSearch={mock_handleSubmit}/>
    )
  });

  it('should start with an inputValue of an empty string', () => {
    expect(wrapper.state('inputValue')).toEqual('');
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state of inputValue on change', () => {
    expect(wrapper.state()).toEqual({ inputValue: '' })
    wrapper.instance()._handleChange({ target: {value: 'new value' }});
    expect(wrapper.state()).toEqual({ inputValue: 'new value' })
  });
});