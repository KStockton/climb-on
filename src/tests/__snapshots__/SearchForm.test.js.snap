// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all the data passed in 1`] = `
<div
  className="SearchForm-header"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="on"
      className="SearchForm-search-bar"
      name="search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
    <button
      className="SearchForm-submit-button"
    >
      <i
        className="fas fa-search"
      />
    </button>
  </form>
</div>
`;
